PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsrFiles
+
+Function Invoke-RequestSignedCertificateByMsca {
+    <#
+        .SYNOPSIS
+        Request CSR to be signed by a MSCA Certificate Authority.
+
+        .DESCRIPTION
+        The Invoke-RequestSignedCertificateByMsca cmdlet connects to MSCA web enrollment to sign a CSR.  
+        
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware"
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io.
+
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware" -getCArootCert
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io and also will retrieve 
+        the Certificate Authority's root chain certificate.  
+
+        .PARAMETER caFqdn
+        The FQDN of the Certificate Authorty web enrollment website.
+
+        .PARAMETER username
+        The username to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER password
+        The password to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER csrFilePath
+        The full file path for the certificate signing request file. 
+
+        .PARAMETER certDir
+        The directory path to store the certificate authority signed certificate file.
+
+        .PARAMETER certificateTemplate
+        The name of the certificate template to be used.  
+
+        .PARAMETER getCArootCert
+        The Certificate Authorty chain root certificate will be retrieved as well.   
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $caFqdn,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $username,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $password,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $csrFilePath,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certDir,            
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certificateTemplate,
+        [Parameter (Mandatory = $false)] [Switch] $getCArootCert
+    )
+
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDir)) {
+        Write-Error "Error: $certDir directory does not exist." -ErrorAction Stop
+    }
+    if (-Not (Test-Path -Path $csrFilePath)) {
+        Write-Error "Error: $csrFilePath file not found." -ErrorAction Stop
```

Comment:
```suggestion
        Write-Error "Certificate signing request file ($csrFilePath ) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53